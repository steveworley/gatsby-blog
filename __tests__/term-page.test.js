import React from 'react'
import renderer from 'react-test-renderer'
import Term from '../src/templates/terms'

import { withPosts, withoutPosts } from '../__mocks__/taxonomy_term__tags.mock'

jest.mock('../src/components/layout',  () => 'Layout')

describe('Taxonomy Term', () => {
  it ('renders correctly with posts', () => {
    const tree = renderer.create(<Term data={withPosts} />).toJSON()
    expect(tree).toMatchSnapshot();
  })

  it ('renders correctly without posts', () => {
    const tree = renderer.create(<Term data={withoutPosts} />).toJSON()
    expect(tree).toMatchSnapshot();
  })
})