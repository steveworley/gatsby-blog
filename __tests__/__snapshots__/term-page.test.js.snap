// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Taxonomy Term renders correctly with posts 1`] = `
<Layout>
  <div>
    <h2>
      Showing posts tagged with: 
      <em>
        CSS
      </em>
    </h2>
    <div>
      <a
        href="youtube-z-index-fix"
      >
        <h3>
          Youtube z-index fix
        </h3>
      </a>
    </div>
    <div>
      <a
        href="css-attribute-selectors"
      >
        <h3>
          CSS attribute Selectors
        </h3>
      </a>
    </div>
  </div>
</Layout>
`;

exports[`Taxonomy Term renders correctly without posts 1`] = `
<Layout>
  <div>
    <h2>
      Showing posts tagged with: 
      <em>
        Youtube
      </em>
    </h2>
  </div>
</Layout>
`;
